packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -248,6 +240,24 @@ export class NgccProcessor {
 
     throw new Error(`Cannot locate the 'node_modules' directory.`);
   }
+
+  private findPackageManagerLockFile(projectBasePath: string): {
+    lockFilePath: string;
+    lockFileData: Buffer;
+  } {
+    for (const lockFile of ['yarn.lock', 'pnpm-lock.yaml', 'package-lock.json']) {
+      const lockFilePath = path.join(projectBasePath, lockFile);
+
+      try {
+        return {
+          lockFilePath,
+          lockFileData: readFileSync(lockFilePath),
+        };
+      } catch {}
+    }
+
+    throw new Error('Cannot locate a package manager lock file.');
```

Comment:
Should we list the files we tried in this error message? Otherwise it's hard to know if the CLI is looking in the wrong place or if the file itself was misplaced.

Reply:
Actually this error message is never shown to the users. As this is non fatal and the exception is caught further down

Reply:
Up to you, could be useful for local development if it's ever thrown for an unexpected reason.